plugin/trino-influxdb/src/main/java/io/trino/plugin/influxdb/InfluxMetadata.java

```diff
@@ -0,0 +1,240 @@
+import com.google.common.collect.ImmutableSet;
+import io.trino.spi.connector.Assignment;
+import io.trino.spi.connector.ColumnHandle;
+import io.trino.spi.connector.ColumnMetadata;
+import io.trino.spi.connector.ConnectorMetadata;
+import io.trino.spi.connector.ConnectorSession;
+import io.trino.spi.connector.ConnectorTableHandle;
+import io.trino.spi.connector.ConnectorTableMetadata;
+import io.trino.spi.connector.Constraint;
+import io.trino.spi.connector.ConstraintApplicationResult;
+import io.trino.spi.connector.LimitApplicationResult;
+import io.trino.spi.connector.ProjectionApplicationResult;
+import io.trino.spi.connector.SchemaTableName;
+import io.trino.spi.connector.SchemaTablePrefix;
+import io.trino.spi.connector.TableColumnsMetadata;
+import io.trino.spi.connector.TableNotFoundException;
+import io.trino.spi.expression.ConnectorExpression;
+import io.trino.spi.predicate.TupleDomain;
+
+import javax.inject.Inject;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.trino.plugin.influxdb.InfluxDataType.toTrinoType;
+import static io.trino.plugin.influxdb.InfluxFilterManager.buildFilter;
+import static io.trino.plugin.influxdb.InfluxLimitManager.buildLimit;
+import static io.trino.plugin.influxdb.InfluxProjectedManager.buildProjection;
+import static java.util.Objects.requireNonNull;
+
+public class InfluxMetadata
+        implements ConnectorMetadata
+{
+    private final InfluxClient client;
+
+    @Inject
+    public InfluxMetadata(InfluxClient client)
+    {
+        this.client = requireNonNull(client, "client is null");
+    }
+
+    @Override
+    public List<String> listSchemaNames(ConnectorSession session)
+    {
+        return ImmutableList.copyOf(client.getSchemaNames());
+    }
+
+    @Override
+    public List<SchemaTableName> listTables(ConnectorSession session, Optional<String> optionalSchemaName)
+    {
+        Set<String> schemaNames = optionalSchemaName.map(ImmutableSet::of)
+                .orElseGet(() -> ImmutableSet.copyOf(client.getSchemaNames()));
+
+        ImmutableList.Builder<SchemaTableName> builder = ImmutableList.builder();
+        for (String schemaName : schemaNames) {
+            for (String tableName : client.getTableNames(schemaName)) {
+                builder.add(new SchemaTableName(schemaName, tableName));
+            }
+        }
+        return builder.build();
+    }
+
+    @Override
+    public InfluxTableHandle getTableHandle(ConnectorSession session, SchemaTableName schemaTableName)
+    {
+        if (!listSchemaNames(session).contains(schemaTableName.getSchemaName())) {
+            return null;
+        }
+
+        Optional<InfluxTable> table = client.getTable(schemaTableName.getSchemaName(), schemaTableName.getTableName());
+        if (table.isEmpty()) {
```

Comment:
We could use functional programming here i.e map and orElse

Reply:
got it. i will check all my codes.